azext_iot/digitaltwins/providers/resource.py

```diff
@@ -76,11 +77,17 @@ def check_state(lro):
                 from time import sleep
                 instance = lro.resource().as_dict()
                 state = instance.get('provisioning_state', None)
-                while state == "Provisioning":
+                retries = 0
+                while state == "Provisioning" and retries < MAX_ADT_CREATE_RETRIES:
+                    retries += 1
                     sleep(1)
                     lro.update_status()
                     instance = lro.resource().as_dict()
                     state = instance.get('provisioning_state', None)
+                if retries == MAX_ADT_CREATE_RETRIES:
+                    raise CLIError(
+                        "Never finished provisioning the Digital Twins instance. Please try again with a different name."
```

Comment:
Just a note - this doesn't necessarily mean that they'll need to create another instance - I think we should just warn users that their resource was created (since the poller returned success / `201`), but that it may not be finished provisioning. Perhaps we could even give them a hand with an example `az dt show -n {name} -g {group}` example command.